src/ui/pages/GetStarted/AddReadonlyAddress/AddReadonlyAddress.tsx

```diff
@@ -0,0 +1,249 @@
+  await setCurrentAddress({ address });
+}
+
+function getHints(
+  query: string,
+  data: undefined | Awaited<ReturnType<(typeof ZerionAPI)['getWalletsMeta']>>
+): { address: string | null; domains: string[] | null } {
+  if (!data || !data.data?.length) {
+    return { address: null, domains: null };
+  }
+
+  if (isEthereumAddress(query)) {
+    return {
+      address: null,
+      domains: data.data[0]?.identities.map((value) => value.handle) || null,
+    };
+  } else {
+    return {
+      address: data.data[0]?.address ?? null,
+      domains: data.data[0]?.identities.map((value) => value.handle) || null,
+    };
+  }
+}
+
+async function lookup(value: string) {
+  const result = await ZerionAPI.getWalletsMeta({ identifiers: [value] });
+  if (result.data?.length === 0) {
+    const message = result.errors?.[0]?.title;
+    throw new Error(message || 'No resolved identities');
+  } else {
+    return result;
+  }
+}
+
+async function lookupAddressByQuery(query: string) {
+  try {
+    const isEthereumAddress = checkIsEthereumAddress(query);
+    const promises = [lookup(query)];
+    if (
+      !isEthereumAddress &&
+      !query.endsWith('.eth') &&
+      !query.endsWith('.lens')
+    ) {
+      promises.push(lookup(`${query}.eth`));
+      promises.push(lookup(`${query}.lens`));
+    }
+    return await Promise.any(promises);
+  } catch (unknownError) {
+    const error =
+      unknownError instanceof AggregateError
+        ? unknownError.errors[0]
+        : unknownError;
+    if (error instanceof HTTPError) {
+      const payload = await (error as HTTPError).response.json();
+      const message = payload.errors?.[0]?.title;
+      if (message) {
+        throw new Error(message);
+      }
+    }
+    throw error;
+  }
+}
+
+export function AddReadonlyAddress() {
+  useBackgroundKind(whiteBackgroundKind);
+  const formId = useId();
+  const navigate = useNavigate();
+
+  const { mutate, isLoading: isSubmitting } = useMutation({
+    mutationFn: submitReadonlyAddress,
+    onSuccess: () => navigate('/overview'),
+  });
+
+  const [debouncedValue, setDebouncedValue] = useState('');
+  const query = debouncedValue.trim();
+  const isEthereumAddress = checkIsEthereumAddress(query);
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  const { data, isInitialLoading, isError, error } = useQuery({
+    queryKey: ['lookupAddressByQuery', query],
+    queryFn: async () => lookupAddressByQuery(query),
+    staleTime: 20000,
+    retry: 0,
+    enabled: Boolean(query),
+    suspense: false,
+  });
+
+  const isDomainResolving = query && isInitialLoading && !isEthereumAddress;
+  const resolveError = isError && !isEthereumAddress;
+
+  useCustomValidity({
+    ref: inputRef,
+    customValidity: isEthereumAddress
+      ? '' // Form MUST be valid for valid eth address regardless of the getWalletsMeta request state
+      : isDomainResolving
+      ? 'Wait until address is resolved'
```

Comment:
I'm not sure that validity here is the best option. I'm getting this "bad format-like" errors after just quick pressing enter and it looks like there is something wrong with the input.

Reply:
It's a question of how to display validity error. What would you suggest should happen when submit button is clicked during loading? Except for disabling button :)

Reply:
I think, the best way is to apply submit with delay in this case.
However, I understand, that this can be a tricky thing to do. 
So as an easy solution, I would probably show some yellow text like "try again after resolving the address"